x-pack/plugins/enterprise_search/server/lib/fetch_indices.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IScopedClusterClient } from 'kibana/server';
+
+import { ByteSizeValue } from '@kbn/config-schema';
+
+import { ElasticsearchIndex } from '../../common/types';
+
+export const fetchIndices = async (client: IScopedClusterClient): Promise<ElasticsearchIndex[]> => {
+  const indexNamesString = 'search-*';
+
+  // This call retrieves alias and settings information about indices
+  const indices = await client.asCurrentUser.indices.get({
+    index: indexNamesString,
+    expand_wildcards: ['open'],
+    // only get specified index properties from ES to keep the response under 536MB
+    // node.js string length limit: https://github.com/nodejs/node/issues/33960
+    filter_path: ['*.aliases'],
+    // for better performance only compute aliases and settings of indices but not mappings
+    // @ts-expect-error new param https://github.com/elastic/elasticsearch-specification/issues/1382
+    features: ['aliases', 'settings'],
+  });
+
+  if (!Object.keys(indices).length) {
+    return [];
+  }
+
+  const { indices: indicesStats = {} } = await client.asCurrentUser.indices.stats({
+    index: indexNamesString,
+    expand_wildcards: ['open'],
+    metric: ['docs', 'store'],
+  });
+  const indicesNames = Object.keys(indices);
+  return indicesNames.map((indexName: string) => {
+    const indexData = indices[indexName];
+    const indexStats = indicesStats[indexName];
+    const aliases = Object.keys(indexData.aliases!);
+    const size_in_bytes = new ByteSizeValue(
```

Comment:
```suggestion
    const sizeInBytes = new ByteSizeValue(
```

Reply:
Minor nit. For consistency with other variable names.

Reply:
Actually it broke the CI but for some reason I cannot get the same error locally 🤔